src/sentry/api/endpoints/chunk_upload.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+
+# from rest_framework.permissions import IsAuthenticated
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.models import FileBlob
+from sentry.models.file import DEFAULT_BLOB_SIZE
+from sentry.api.base import Endpoint
+from sentry.api.bases.project import ProjectReleasePermission
+
+
+UPLOAD_ENDPOINT = 'https://sentry.io'
+MAX_CHUNKS_PER_REQUEST = 16
+MAX_CONCURRENCY = 4
+HASH_ALGORITHM = 'sha1'
+
+
+class ChunkUploadEndpoint(Endpoint):
+    permission_classes = (ProjectReleasePermission, )
+
+    def get(self, request):
+        return Response(
+            {
+                'url': UPLOAD_ENDPOINT,
+                'chunkSize': DEFAULT_BLOB_SIZE,
+                'chunksPerRequest': MAX_CHUNKS_PER_REQUEST,
+                'concurrency': MAX_CONCURRENCY,
+                'hashAlgorithm': HASH_ALGORITHM,
+            }
+        )
+
+    def post(self, request):
+        files = request.FILES.getlist('file')
+
+        if len(files) > MAX_CHUNKS_PER_REQUEST:
+            return Response({'error': 'Too many chunks'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        elif len(files) == 0:
+            # No files uploaded is ok
+            return Response(status=status.HTTP_200_OK)
+
+        # Validate file size
+        checksum_list = []
+        for chunk in files:
+            if chunk._size > DEFAULT_BLOB_SIZE:
+                return Response({'error': 'Chunk size too large'},
+                                status=status.HTTP_400_BAD_REQUEST)
+            checksum_list.append(chunk._name)
+
+        for chunk in files:
+            # Here we create the actual file
+            blob = FileBlob.from_file(chunk)
+            if blob.checksum not in checksum_list:
+                # We do not clean up here since we have a cleanup job
+                return Response({'error': 'Checksum missmatch'},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(status=status.HTTP_200_OK)
```

Comment:
nit: Shouldn't this be `201 Created`?

Reply:
We've discussed this and we went with 200.
But I don't have a strong opinion we can go with 201.

Reply:
Ahh and also, if the endpoint already has chunks, they will not get "created" because they are already there.

Reply:
In case of a single chunk per request (see previous comment), we could even go with `PUT` and `200 OK`.

Reply:
That's fine.

Reply:
So I will leave it like this, since a `PUT` would mean a new function and 201 will not always be correct?! 🤷‍♂️